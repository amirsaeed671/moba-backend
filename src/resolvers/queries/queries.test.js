const mongoose = require('mongoose');
const {gql} = require('apollo-server-express');
const CreateTestServer = require('../../test-server');
const {User, Hero} = require('../../mongodb/models');

const ALL_USERS = gql`
    query {
        allUsers {
            name
        }
    }
`;

describe("queries test", () => {

    beforeAll(async () => {
        await mongoose.connect(process.env.MONGO_URL, {useNewUrlParser: true, useUnifiedTopology: true, useCreateIndex: true, useFindAndModify: false}, (err) => {
            if (err) {
                console.error(err);
                process.exit(1);
            }
        });
    });


    test("initial test", async () => {
        const find = jest.fn()

        const {query} = CreateTestServer({
            User
        })

        const res = await query({query: ALL_USERS});

        expect(res).toMatchSnapshot();
    });
});