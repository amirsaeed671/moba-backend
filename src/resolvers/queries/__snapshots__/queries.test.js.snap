// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queries test initial test 1`] = `
Object {
  "data": Object {
    "allUsers": Array [],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
